### PR TITLE
Engine - Adapt Syslog decoder to be an integration

### PR DESCRIPTION
**Pull request by nmkoremblum**
|Related issue|
|---|
|#17160|

## Description

As requested on issue #17160, Syslog decoder was adapted to be an integration.

## Logs/Alerts example

```syslog
Oct 11 22:14:15 mymachine su: 'su root' failed for lonvick on /dev/pts/8

Feb  5 17:32:18 10.0.0.99 Use the BFG!

Aug 24 05:34:00 CST 1987 mymachine myproc[10]: %% It's time to make the do-nuts. %%  Ingredients: Mix=OK, Jelly=OK # Devices: Mixer=OK, Jelly_Injector=OK, Frier=OK # Transport: Conveyer1=OK, Conveyer2=OK # %%

Oct 22 10:52:12 scapegoat 1990 Oct 22 10:52:01 TZ-6 scapegoat.dmz.example.org 10.1.2.3 sched[0]: That's All Folks!

2018-08-14T14:30:02.203151+02:00 linux-sqrz systemd[4179]: Stopped target Basic System.

1990 Oct 22 10:52:01 TZ-6 scapegoat.dmz.example.org 10.1.2.3 sched[0]: That's All Folks!
```

## Tests

```json
[
    {
        "wazuh": {
            "queue": 49,
            "location": "/dev/stdin",
            "message": "Oct 11 22:14:15 mymachine su: 'su root' failed for lonvick on /dev/pts/8",
            "decoders": [
                "core-wazuh-message",
                "core-localfile",
                "integrations",
                "syslog"
            ],
            "source": "logcollector"
        },
        "event": {
            "original": "Oct 11 22:14:15 mymachine su: 'su root' failed for lonvick on /dev/pts/8",
            "start": "2023-10-11T22:14:15.000Z",
            "kind": "event"
        },
        "host": {
            "hostname": "mymachine"
        },
        "~TAG": "su",
        "message": "'su root' failed for lonvick on /dev/pts/8",
        "related": {
            "hosts": [
                "mymachine"
            ]
        },
        "process": {
            "name": "su"
        }
    },
    {
        "wazuh": {
            "queue": 49,
            "location": "/dev/stdin",
            "message": "Feb  5 17:32:18 10.0.0.99 Use the BFG!",
            "decoders": [
                "core-wazuh-message",
                "core-localfile",
                "integrations",
                "syslog"
            ],
            "source": "logcollector"
        },
        "event": {
            "original": "Feb  5 17:32:18 10.0.0.99 Use the BFG!",
            "start": "2023-02-05T17:32:18.000Z",
            "kind": "event"
        },
        "host": {
            "hostname": "10.0.0.99"
        },
        "message": "Use the BFG!",
        "related": {
            "hosts": [
                "10.0.0.99"
            ]
        }
    },
    {
        "wazuh": {
            "queue": 49,
            "location": "/dev/stdin",
            "message": "Aug 24 05:34:00 CST 1987 mymachine myproc[10]: %% It's time to make the do-nuts. %%  Ingredients: Mix=OK, Jelly=OK # Devices: Mixer=OK, Jelly_Injector=OK, Frier=OK # Transport: Conveyer1=OK, Conveyer2=OK # %%",
            "decoders": [
                "core-wazuh-message",
                "core-localfile",
                "integrations",
                "syslog"
            ],
            "source": "logcollector"
        },
        "event": {
            "original": "Aug 24 05:34:00 CST 1987 mymachine myproc[10]: %% It's time to make the do-nuts. %%  Ingredients: Mix=OK, Jelly=OK # Devices: Mixer=OK, Jelly_Injector=OK, Frier=OK # Transport: Conveyer1=OK, Conveyer2=OK # %%",
            "start": "2023-08-24T05:34:00.000Z",
            "kind": "event"
        },
        "host": {
            "hostname": "CST"
        },
        "message": "1987 mymachine myproc[10]: %% It's time to make the do-nuts. %%  Ingredients: Mix=OK, Jelly=OK # Devices: Mixer=OK, Jelly_Injector=OK, Frier=OK # Transport: Conveyer1=OK, Conveyer2=OK # %%",
        "related": {
            "hosts": [
                "CST"
            ]
        }
    },
    {
        "wazuh": {
            "queue": 49,
            "location": "/dev/stdin",
            "message": "Oct 22 10:52:12 scapegoat 1990 Oct 22 10:52:01 TZ-6 scapegoat.dmz.example.org 10.1.2.3 sched[0]: That's All Folks!",
            "decoders": [
                "core-wazuh-message",
                "core-localfile",
                "integrations",
                "syslog"
            ],
            "source": "logcollector"
        },
        "event": {
            "original": "Oct 22 10:52:12 scapegoat 1990 Oct 22 10:52:01 TZ-6 scapegoat.dmz.example.org 10.1.2.3 sched[0]: That's All Folks!",
            "start": "2023-10-22T10:52:12.000Z",
            "kind": "event"
        },
        "host": {
            "hostname": "scapegoat"
        },
        "message": "1990 Oct 22 10:52:01 TZ-6 scapegoat.dmz.example.org 10.1.2.3 sched[0]: That's All Folks!",
        "related": {
            "hosts": [
                "scapegoat"
            ]
        }
    },
    {
        "wazuh": {
            "queue": 49,
            "location": "/dev/stdin",
            "message": "2018-08-14T14:30:02.203151+02:00 linux-sqrz systemd[4179]: Stopped target Basic System.",
            "decoders": [
                "core-wazuh-message",
                "core-localfile",
                "integrations",
                "syslog"
            ],
            "source": "logcollector"
        },
        "event": {
            "original": "2018-08-14T14:30:02.203151+02:00 linux-sqrz systemd[4179]: Stopped target Basic System.",
            "start": "2018-08-14T12:30:02.203Z",
            "kind": "event"
        },
        "host": {
            "hostname": "linux-sqrz"
        },
        "~TAG": "systemd",
        "process": {
            "pid": 4179,
            "name": "systemd"
        },
        "message": "Stopped target Basic System.",
        "related": {
            "hosts": [
                "linux-sqrz"
            ]
        }
    },
    {
        "wazuh": {
            "queue": 49,
            "location": "/dev/stdin",
            "message": "1990 Oct 22 10:52:01 TZ-6 scapegoat.dmz.example.org 10.1.2.3 sched[0]: That's All Folks!",
            "decoders": [
                "core-wazuh-message",
                "core-localfile",
                "integrations",
                "syslog"
            ],
            "source": "logcollector"
        },
        "event": {
            "original": "1990 Oct 22 10:52:01 TZ-6 scapegoat.dmz.example.org 10.1.2.3 sched[0]: That's All Folks!",
            "start": "1990-10-22T10:52:01.000Z",
            "kind": "event"
        },
        "~timezone": "TZ-6",
        "host": {
            "hostname": "scapegoat.dmz.example.org",
            "ip": "10.1.2.3"
        },
        "~TAG": "sched",
        "process": {
            "pid": 0,
            "name": "sched"
        },
        "message": "That's All Folks!",
        "related": {
            "hosts": [
                "scapegoat.dmz.example.org"
            ]
        }
    }
]
```

## Tasks
- [x] Adapt decoders
  - [x] Syslog
- [x] Update Documentation

### CODE REVIEW

**Review by cborla — APPROVED** [review]
LGTM!

File: src/engine/ruleset/integrations/syslog/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  Please, see https://www.ietf.org/rfc/rfc3164.txt.
+configuration: >
+  For example, a localfile can be added to the `ossec.conf` file in a monitored agent as in the following block:
+
+  ```xml
+  <localfile>
+    <!-- Edit location to appropriate path if needed -->
+    <location>/tmp/syslog_formatted.log</location>
+    <log_format>json</log_format>
+    <label key="event.module">syslog</label>
```
Comment (JavierBejMen) [inline src/engine/ruleset/integrations/syslog/documentation.yml:14]: Comment this block, we are not using tags. And log format should be syslog.

**Review by JavierBejMen — CHANGES_REQUESTED** [review]

File: src/engine/ruleset/integrations/syslog/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+## Configuration
+
+For example, a localfile can be added to the `ossec.conf` file in a monitored agent as in the following block:
+```xml <localfile>
+  <!-- Edit location to appropriate path if needed -->
+  <location>/tmp/syslog_formatted.log</location>
+  <log_format>json</log_format>
+  <label key="event.module">syslog</label>
+  <label key="event.dataset">syslog.log</label>
+</localfile> ```
```
Comment (JcabreraC) [inline src/engine/ruleset/integrations/syslog/README.md:24]:
```suggestion
```xml <localfile>
  <!-- Edit location to appropriate path if needed -->
  <location>/tmp/syslog_formatted.log</location>
  <log_format>syslog</log_format>
</localfile> ```
```

File: src/engine/ruleset/integrations/syslog/documentation.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    <label key="event.dataset">syslog.log</label>
+  </localfile>
+  ```
```
Comment (JcabreraC) [inline src/engine/ruleset/integrations/syslog/documentation.yml:17]:
```suggestion
  ```xml
  <localfile>
    <!-- Edit location to appropriate path if needed -->
    <location>/tmp/syslog_formatted.log</location>
    <log_format>syslog</log_format>
  </localfile>
  ```
```

**Review by JcabreraC — CHANGES_REQUESTED** [review]

File: src/engine/ruleset/integrations/syslog/decoders/syslog.yml [hunk 4]
```diff
@@ -12,12 +12,11 @@ metadata:
     url: https://wazuh.com
     date: 2022/11/08
   references:
-    - https://datatracker.ietf.org/doc/html/rfc3164
-    - https://datatracker.ietf.org/doc/html/rfc5424
+    - https://ietf.org/doc/html/rfc3164
+    - https://ietf.org/doc/html/rfc5424
```
Comment (JcabreraC) [inline src/engine/ruleset/integrations/syslog/decoders/syslog.yml:16]: Code: 404 Page Not Found

**Review by JcabreraC — CHANGES_REQUESTED** [review]

**Review by JavierBejMen — APPROVED** [review]
LGTM

**Review by JcabreraC — CHANGES_REQUESTED** [review]
LGTM !

**Review by JcabreraC — APPROVED** [review]

**Pull request merged**